test/system/src/main/java/io/pravega/test/system/framework/services/docker/DockerBasedService.java

```diff
@@ -0,0 +1,198 @@
+                log.info("Network addresses {}", taskList.get(0).networkAttachments().get(0).addresses().get(0));
+                List<Service> serviceList = Exceptions.handleInterrupted(() -> dockerClient.listServices(criteria));
+                log.info("Service list size {}", serviceList.size());
+                for (int i = 0; i < taskList.size(); i++) {
+                    log.info("task {}", taskList.get(i).name());
+                    if (taskList.get(i).status().state().equals(TaskStatus.TASK_STATE_RUNNING)) {
+                        String[] uriArray = taskList.get(i).networkAttachments().get(0).addresses().get(0).split("/");
+                        ImmutableList<PortConfig> numPorts = Exceptions.handleInterrupted(() -> dockerClient.inspectService(serviceList.get(0).id()).endpoint().spec().ports());
+                        for (int k = 0; k < numPorts.size(); k++) {
+                            int port = numPorts.get(k).publishedPort();
+                            log.info("Port {}", port);
+                            log.info("Uri list {}", uriArray[0]);
+                            URI uri = URI.create("tcp://" + uriArray[0] + ":" + port);
+                            uriList.add(uri);
+                        }
+
+                    }
+                }
+            }
+        } catch (DockerException e) {
+            log.error("Unable to list service details", e);
+        }
+        return uriList;
+    }
+
+    @Override
+    public CompletableFuture<Void> scaleService(final int instanceCount) {
+        try {
+            Preconditions.checkArgument(instanceCount >= 0, "negative value: %s", instanceCount);
+            Service.Criteria criteria = Service.Criteria.builder().serviceName(this.serviceName).build();
+            TaskSpec taskSpec = Exceptions.handleInterrupted(() -> dockerClient.listServices(criteria).get(0).spec().taskTemplate());
+            String serviceId = Exceptions.handleInterrupted(() -> dockerClient.listServices(criteria).get(0).id());
+            EndpointSpec endpointSpec = Exceptions.handleInterrupted(() -> dockerClient.inspectService(serviceId).spec().endpointSpec());
+            Service service = Exceptions.handleInterrupted(() -> dockerClient.inspectService(serviceId));
+            Exceptions.handleInterrupted(() -> dockerClient.updateService(serviceId, service.version().index(), ServiceSpec.builder().endpointSpec(endpointSpec).mode(ServiceMode.withReplicas(instanceCount)).taskTemplate(taskSpec).name(serviceName).build()));
+            String updateState = Exceptions.handleInterrupted(() -> dockerClient.inspectService(serviceId).updateStatus().state());
+            log.info("Update state {}", updateState);
+            return Exceptions.handleInterrupted(() -> waitUntilServiceRunning());
+        } catch (DockerException e) {
+            throw new TestFrameworkException(TestFrameworkException.Type.RequestFailed, "Unable to scale service to given instances.Test Failure", e);
```

Comment:
Can we add more detail to the exception message? In particular, about `given instances`. Also, add a colon like this: `... Test failure: `.

Reply:
done.